transform-message-formats/README.md

```diff
@@ -0,0 +1,495 @@
+
+### Step 2: Develop the service
+
+Follow the instructions given in this section to develop the service.
+
+1. Create a new Ballerina project using the `bal` command and open it in VS Code.
+
+```bash
+$ bal new transform-message-formats
+```
+
+2. Introduce the source code in files with the `.bal` extension (e.g., the `main.bal` file).
+
+Import the
+- `ballerina/http` module to develop the REST API and define the clients that can be used to send requests to the backend services
+- `ballerina/log` module to log debug, error, or info level information for each client request
+
+```ballerina
+import ballerina/http;
+import ballerina/log;
+```
+
+3. Define two [configurable variables](https://ballerina.io/learn/by-example/#configurability) for the port on which the listener should listen and the URL of the hospital service.
+
+```ballerina
+configurable int port = 8290;
+configurable string hospitalServiceUrl = "http://localhost:9090";
+```
+
+4. Define an [`http:Client`](https://ballerina.io/learn/by-example/#http-client) client to send requests to the backend hospital services.
+
+```ballerina
+final http:Client hospitalServiceEP = check initializeHttpClient();
+
+function initializeHttpClient() returns http:Client|error => new (hospitalServiceUrl);
+```
+
+> **Note** The argument passed to the `new` expression (`hospitalServiceUrl`) is the URL of the backend hospital service.
+> 
+> Here, a separate function is used to initialize the clients to aid with testing. Alternatively, the `new` expression can be used directly to initialize the clients.
+>
+> ```ballerina
+> final http:Client hospitalServiceEP = new (hospitalServiceUrl);
+>```
+
+5. Use the data-mapper to define the `transform` function which transform the payload from `HealthcareReservation` to `HospitalReservation`. 
+
+    Follow the guide shown below to create the `transform` function.
+
+    ![Data mapper guide](./resources/tutorial_transform_message_formats_data_mapper_guide.gif)
+
+    The data mapper view of the completed `transform` function is shown below.
+
+    ![Data mapper view](./resources/tutorial_transform_message_formats_data_mapper_view.png)
+
+    The source code of the `transform` function will be as follows.
+
+    ```ballerina
+    isolated function transform(HealthcareReservation healthcareReservation) returns HospitalReservation => {
+    patient: {
+        name: healthcareReservation.name,
+        dob: healthcareReservation.dob,
+        ssn: healthcareReservation.ssn,
+        address: healthcareReservation.address,
+        phone: healthcareReservation.phone,
+        email: healthcareReservation.email
+    },
+    doctor: healthcareReservation.doctor,
+    hospital: healthcareReservation.hospital,
+    appointment_date: healthcareReservation.appointment_date
+};
+    ```
+
+6. Define rest of the records corresponding to the response payloads. The final set of records will look like this.
+
```

Comment:
I would not include the rest of the records (already generated ones) here. Since they will be available in the complete source section anyway.